src/main/java/uk/gov/hmcts/reform/sscs/config/HmcHearingsEventTopicListener.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.sscs.config;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusException;
+import com.azure.messaging.servicebus.ServiceBusFailureReason;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.sscs.model.HmcMessage;
+
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@Slf4j
+@Data
+@Configuration
+public class HmcHearingsEventTopicListener {
+
+    @Value("${azure.service-bus.connectionString}")
+    private String connectionString;
+    @Value("${azure.service-bus.topicName}")
+    private String topicName;
+    @Value("${azure.service-bus.subscriptionName}")
+    private String subscriptionName;
+
+    //TODO add @Bean and add correct values (connectionString,topicName,subscriptionName) in application.yml
+    @SuppressWarnings({"PMD.CloseResource"})
+    public void receiveMessages() {
+        CountDownLatch countdownLatch = new CountDownLatch(1);
+
+        ServiceBusProcessorClient processorClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .processor()
+            .topicName(topicName)
+            .subscriptionName(subscriptionName)
+            .processMessage(HmcHearingsEventTopicListener::processMessage)
+            .processError(context -> processError(context, countdownLatch))
+            .buildProcessorClient();
+
```

Comment:
I think we should create our own retry options with configurable options. The ServiceBusProcessorClient defaults to Retry options that look like:

` public AmqpRetryOptions() {
        this.maxRetries = 3;
        this.delay = Duration.ofMillis(800);
        this.maxDelay = Duration.ofMinutes(1);
        this.tryTimeout = Duration.ofMinutes(1);
        this.retryMode = AmqpRetryMode.EXPONENTIAL;
    }`

It would be beneficial to make the delay, max retry and timeout configurable.

Reply:
Zikrur, I have done this on my branch for reference https://github.com/hmcts/sscs-hearings-api/pull/83 in the class HearingsEventQueueListener

Reply:
added.